File: go/problems/simplified_fractions.go
package problems

import "fmt"

func SimplifiedFractions(n int) []string {
  var results []string
  if n == 1 {
    return results
  }
  for i := 1; i <= n; i++ {
    for j := i + 1; j <= n; j++ {
      if gcd(j, i) == 1 {
        fraction := fmt.Sprintf("%d/%d", i, j)
        results = append(results, fraction)
      }
    }
  }
  return results
}

func gcd(a, b int) int {
  for b != 0 {
    a, b = b, a%b
  }
  return a
}

File: rust/src/problems/simplified_fractions.rs
pub fn simplified_fractions(n: i32) -> Vec<String> {
  let mut results: Vec<String> = Vec::new();
  if n == 1 {
    return results;
  }
  for i in 1..=n {
    for j in i + 1..=n {
      println!("{}, {}, {}", i, j, gcd(j, i));
      if gcd(j, i) == 1 {
        let fraction: String = format!("{}/{}", i, j);
        results.push(fraction);
      }
    }
  }
  return results;
}

fn gcd(mut a: i32, mut b: i32) -> i32 {
  while b != 0 {
    a = b;
    b = a % b;
  }
  return a;
}